File: Difficulty: Easy/Pattern Searching/pattern-searching.js
//{ Driver Code Starts
//Initial Template for javascript

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });

    main();
});

function readLine() {
    return inputString[currentLine++];
}


function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {
        let [text, pat] = readLine().trim().split(" ");
        let obj = new Solution();
        let res = obj.search(text, pat);
        console.log(res);
    
console.log("~");
}
}

// } Driver Code Ends

// User function Template for javascript

/**
 * @param {string} text
 * @param {string} pat
 * @return {number}
 */

class Solution {
    // Function to check if the pattern exists in the text
    search(text, pat) {
        // Create a regular expression object for the pattern with global search option
        let regex = new RegExp(pat, "g");
        
        // Check if the pattern exists in the text using the regular expression
        if (regex.test(text)) {
            // If the pattern is found, return 1
            return 1;
        } else {
            // If the pattern is not found, return 0
            return 0;
        }
    }
}

File: Difficulty: Easy/Pattern searching/pattern-searching.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

// User function Template for C++

/*
str: string in which pattern we have to find pattern.
pat: pattern needs to searched.
*/

bool searchPattern(string &txt, string &pat) {
    // Using the find() function to check if the pattern exists in the string
    return txt.find(pat) != string::npos;  // Returns true if pattern is found, false otherwise
}


//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;

    while (t--) {
        string str, pat;
        cin >> str;
        cin >> pat;

        if (searchPattern(str, pat) == true)
            cout << "true" << endl;
        else
            cout << "false" << endl;

        cout << "~"
             << "\n";
    }
}

// } Driver Code Ends